src/deploy/functions/release/fabricator.ts

```diff
@@ -214,7 +214,38 @@ export class Fabricator {
   }
 
   async deleteEndpoint(endpoint: backend.Endpoint): Promise<void> {
-    await this.deleteTrigger(endpoint);
+    try{
+      await this.deleteTrigger(endpoint);
+    }
+    catch(err) {
+      /**
+       * In cases where a function has been partially deleted, 
+       * we may encounter errors on subsequent attempts to delete it.
+       * 
+       * For example, if an account lacking the `pubsub.topics.delete` permission
+       * attempts to delete a function with a Pub/Sub trigger, the trigger will be
+       * deleted but the topic will not. This will cause the function deletion
+       * to fail on subsequent attempts.
+       * 
+       */
+      if (
+          err instanceof reporter.DeploymentError && 
+          err.op == "delete schedule" && 
+          (err.original as FirebaseError).message == "HTTP Error: 404, Job not found."
+        ) {
+          logger.warn(`Trigger for ${endpoint.id} not found. Continuing to delete function.`);
+      }
+      else if (
+          err instanceof reporter.DeploymentError && 
+          err.op == "delete topic" && 
+          err.message.startsWith("Failed to delete topic function")
```

Comment:
This concerns me - if the topic deletion fails for a generic reason, I don't like orphaning the resource. If we can check for specific failures (ie 404s), I'd be more comfortable with this